transport/rabbitmq/src/test/kotlin/RabbitMqTestContainer.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2023 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.transport.rabbitmq
+
+import io.kotest.core.extensions.install
+import io.kotest.core.spec.Spec
+import io.kotest.extensions.testcontainers.ContainerExtension
+
+import org.eclipse.apoapsis.ortserver.config.ConfigManager
+import org.eclipse.apoapsis.ortserver.config.ConfigSecretProviderFactoryForTesting
+import org.eclipse.apoapsis.ortserver.transport.testing.createConfigManager
+
+import org.testcontainers.containers.RabbitMQContainer
+
+/**
+ * Extension function to start an RabbitMq broker in a test container for testing a factory of the given [consumerName]
+ * and [transportType]. The resulting [ConfigManager] can be used to connect to the broker in the container.
+ */
+fun Spec.startRabbitMqContainer(consumerName: String, transportType: String): ConfigManager {
+    val rabbitMq = install(
+        ContainerExtension(
+            RabbitMQContainer("rabbitmq")
+        )
+    )
```

Comment:
How does it work, does Kotest automatically stop the container after the test if it is installed inside a test?

Reply:
Here's the source for [ContainerExtension](https://github.com/kotest/kotest-extensions-testcontainers/blob/master/src/main/kotlin/io/kotest/extensions/testcontainers/ContainerExtension.kt).

Looks like the default lifecycle is that the installed extension is active for the rest of the project. You might want to set `mode = ContainerLifecycleMode.Spec`

Reply:
Thanks @Kantis for chiming in! However, I believe the default `Project` mode is the right one to use right now because

> This mode can be used with multiple test containers by using separate instances of the container extensions.

and we *are* currently using multiple test container instances in a single spec.